frontend/out/production/frontend/frontend

```diff
@@ -0,0 +1,308 @@
+        } catch (Exception ex) {
+            return ex.getMessage();
+        }
+        return getResults();
+    }
+
+    public String createExchangeItem(String customerID, String newItemID, String oldItemID, String dateOfExchange) {
+        String userPrefix = customerID.substring(0, 2);
+        ClientRequest request = null;
+        if(userPrefix.equalsIgnoreCase("qc")) {
+            request = new ClientRequest(OperationCode.EXCHANGE_ITEM, Location.QUEBEC, UserType.MANAGER);
+        }
+        if(userPrefix.equalsIgnoreCase("bc")){
+            request = new ClientRequest(OperationCode.EXCHANGE_ITEM, Location.BRITISHCOLUMBIA, UserType.MANAGER);
+        }
+        if(userPrefix.equalsIgnoreCase("on")){
+            request = new ClientRequest(OperationCode.EXCHANGE_ITEM, Location.ONTARIO, UserType.MANAGER);
+        }
+
+        request.addRequestDataEntry(ParameterType.CLIENTID, customerID);
+        request.addRequestDataEntry(ParameterType.NEWITEMID, newItemID);
+        request.addRequestDataEntry(ParameterType.OLDITEMID, oldItemID);
+        request.addRequestDataEntry(ParameterType.DATEOFEXCHANGE, dateOfExchange);
+
+        try {
+            sendRequestToSequencer(OperationCode.EXCHANGE_ITEM, request);
+        } catch (Exception ex) {
+            return ex.getMessage();
+        }
+        return getResults();
+    }
+
+
+    @Override
+    public void softwareFailure(String managerID) {
+        requiredAnswersForAgreement--;
+    }
+
+    @Override
+    public void replicaCrash(String managerID) {
+        requiredAnswersForAgreement--;
+    }
+
+
+    private void sendRequestToSequencer(OperationCode method, ClientRequest clientRequest) throws Exception {
+        ByteArrayOutputStream bStream = new ByteArrayOutputStream();
+        ObjectOutput oo = new ObjectOutputStream(bStream);
+        oo.writeObject(clientRequest);
+        oo.close();
+        oo.flush();
+
+        byte[] serializedClientRequest = bStream.toByteArray();
+        String payload = new String(serializedClientRequest);
+
+        MessageRequest messageToSend = null;
+        try {
+            messageToSend = new MessageRequest(method, 0, payload, EntityAddressBook.SEQUENCER);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        if (messageToSend != null) {
+            if (!socket.send(messageToSend, 5, 1000)) {
+                throw new Exception("Unable to process message within system!");
+            }
+        }
+    }
+
+    private String getResults(){
+        int seqNumber = Integer.valueOf(socket.getResponse().getOperationParameters().substring("SEQ=".length()));
+
+        ConsensusTracker tracker = new ConsensusTracker(requiredAnswersForAgreement, seqNumber);
+
+        frontendListener.setTracker(tracker);
+
+        try {
+            tracker.Wait();
+        } catch (InterruptedException ex) {
+            return "Error: could not obtain answer";
```

Comment:
if youre using semaphores, you need to put this into a while loop that continuously tries to grap the lock, otherwise it just fails and you neveer get any results. Also a semaphore may not be necessary, this thread seems to be running iteratively